src/control/lib/spdk/include/nvme_control.h

```diff
@@ -50,4 +50,17 @@ nvme_format(char *ctrlr_pci_addr);
 struct ret_t *
 nvme_fwupdate(char *ctrlr_pci_addr, char *path, unsigned int slot);
 
+/**
+ * Initialize SPDK environment.
+ *
+ * \param mem_sz size of memory allocated to environment (mb).
+ * \param env_ctx environment context string (DPDK).
+ * \param nr_pcil size of pcil.
+ * \param pcil list of allowed PCI addresses of NVMe controllers.
+ *
+ * \return a pointer to a return struct (ret_t).
+ */
+struct ret_t *
+daos_spdk_init(int mem_sz, char *env_ctx, size_t nr_pcil, char **pcil);
```

Comment:
Hmm. Wouldn't it be better to continue to use the opts struct so that this signature doesn't need to be modified every time a new version of SPDK/DPDK introduces some new configuration requirement?

Reply:
recorded as part of https://jira.hpdd.intel.com/browse/DAOS-7960 to be done as a follow-up task